lib/file/data.dart

```diff
@@ -6,29 +6,42 @@ import 'package:time_machine/time_machine.dart';
 
 part 'data.g.dart';
 
-@HiveType(typeId: typeFile)
-class File implements Entity, Comparable<File> {
+@HiveType(typeId: TypeId.typeFile)
+class File implements Entity<File>, Comparable<File> {
   File({
     @required this.id,
     @required this.name,
-    @required this.mimeType,
     @required this.owner,
     @required this.createdAt,
     @required this.updatedAt,
-    @required this.isDirectory,
     @required this.parent,
-    this.size,
+    @required this.isDirectory,
+    @required this.mimeType,
+    @required this.size,
   })  : assert(id != null),
         assert(name != null),
         assert(owner != null),
         assert(owner is Id<User> || owner is Id<Course>),
-        assert(isDirectory != null);
+        files = LazyIds<File>(
+          collectionId: 'files in directory $id',
+          fetcher: () async {
+            final jsonData = await fetchJsonListFrom(
+              'fileStorage',
+              wrappedInData: false,
+              parameters: {
+                'owner': owner.value,
+                'parent': id.toString(),
+              },
+            );
+            return File.fromJsonList(jsonData);
```

Comment:
We could reuse `fetchByOwner` by renaming it to `fetchList` and adding an optional `parentId` attribute